packages/app/src/cli/models/app/app.ts

```diff
@@ -241,6 +244,22 @@ export class App implements AppInterface {
     this.nodeDependencies = nodeDependencies
   }
 
+  async preDeployValidation() {
+    const functionExtensionsWithUiHandle = this.allExtensions
+      .filter((ext) => ext.isFunctionExtension)
+      .map((ext) => ext as unknown as ExtensionInstance<FunctionConfigType>)
+      .filter((ext) => ext.configuration.ui?.handle)
```

Comment:
The extra `map` and `filter` calls add some runtime overhead (because it's iterating over the list multiple times). Even though it's small, we should probably still remove them since they're only here for TypeScript 😛 

Could we do something like this?

```suggestion
      .filter((ext) => ext.isFunctionExtension && (ext.configuration as unknown as FunctionConfigType).ui?.handle)
```